### ISSUE TITLE
Keyboard events issue

### ISSUE THREAD

**Issue by excitoon** [Genymobile/scrcpy#1748]
I'm not sure if that's SDL or of this project, bug, but:

On Linux/Ubuntu, when you start application with unicode keyboard layout, KEYUP/KEYDOWN events contain keycodes (both sym and scancode) which are >256, even if you change layout while application is running. However if you run application having english layout selected, even if you change layout, you only get codes <256 in KEYUP/KEYDOWN. That leads to errors when sometimes your input is going to screen of the device 2 times per key press, and sometimes - 0 times.

TEXT events contain right symbols, so `prefer-text` would be good workaround.

@rom1v do you handle this SDL thing correctly? Is that their known behavior?

```
$ sdl2-config --version
2.0.8
```

**Comment by rom1v** [comment]
> However if you run application having english layout selected, even if you change layout, you only get codes <256 in KEYUP/KEYDOWN.

For keycodes, it seems expected:

https://github.com/spurious/SDL-mirror/blob/66e0e5ad0d554d88d58b92c4ba70d1e4b2f98282/include/SDL_keycode.h#L47-L48

https://github.com/spurious/SDL-mirror/blob/66e0e5ad0d554d88d58b92c4ba70d1e4b2f98282/include/SDL_keycode.h#L154

> do you handle this SDL thing correctly

scrcpy just receives the SDL key events `SDL_KEYUP` and `SDL_KEYDOWN`:

https://github.com/Genymobile/scrcpy/blob/0be766e71a30eec1709da7b24999b6df2b40dc65/app/src/scrcpy.c#L200-L204

**Comment by excitoon** [comment]
@rom1v I'm not about that thing, I see here that SDL application sticks to that keyboard layout with which it was started o_O, for `KEY` events, if I start with Russian then I receive Russian unicode characters no matter how I change it during that run of the application. This is totally unexpected, and good thing is that normally `KEY` events are not used for text processing.

**Comment by rom1v** [comment]
> I see here that SDL application sticks to that keyboard layout with which it was started o_O, for KEY events, if I start with Russian then I receive Russian unicode characters no matter how I change it during that run of the application.

Yes, when I wrote "scrcpy just receives the SDL key events `SDL_KEYUP` and `SDL_KEYDOWN`", I meant that scrcpy just react to what SDL sends. So if SDL does not react to keyboard layout changes, I'm not sure scrcpy could do something.

Ref: #1729

**Comment by excitoon** [comment]
I guess we shall never process text characters in `input_manager_process_key` in order to have consistent behavior :man_shrugging: No matter `prefer-text` is set or not.

**Comment by excitoon** [comment]
@rom1v I guess we can tune SDL somehow. For example for X11 they call `X11_UpdateKeymap` in order to load keymap from X11 to SDL. It's likely that this method is just skipping being called for some reason, when user switches his keymap. It can be `scrcpy` bug, really.
